### PR TITLE
Add initial set of suggested alerts

### PR DESCRIPTION
**Pull request by douglascamata**
* Alerts that might fire because of how tenants define their rules or Observatorium credentials are all set to `severity: "warning"`. 
* Alerts that might fire because of problems in `obsctl-reloader` itself are set to `severity: "critical"`. Examples: listing the `PrometheusRule` CRDs in the cluster and communicating with Observatorium's rules API.

Looking for a good review of the summary and descriptions, particularly for the string templating there.

### CODE REVIEW

File: openshift/alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    expr: |
+      (
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"403", job="obsctl-reloader"}[5m])
+      /
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"2..", job="obsctl-reloader"}[5m])
+      ) or vector(0)
+      > 0.10
+    for: 10m
+    labels:
+      severity: warning
```
Comment (coleenquadros) [inline openshift/alerts.yaml:45]: What if multiple tenants are not able to authenticate, would we still keep it at warning?

Comment (douglascamata) [reply openshift/alerts.yaml:45]: Yes. Unfortunately there's no good solution here that will be automatically good for every setup. It's even not possible to know in advance how many tenants there are in total to know whether it's 10%, 20%, or 100% of the tenants with issues (what would be the threshold to consider it more severe?). This is mostly an informative alert.

Plus authentication verification should live where it's verified, in Observatorium API.

**Review by thibaultmg — APPROVED** [review]
Looks good!

**Review by philipgough — APPROVED** [review]

**Comment by douglascamata** [comment]
I removed the alert `ObsCtlRulesStoreAuthenticationError`, that would always trigger when `ObsCtlRulesStoreServerError` triggers, as it matches on any 5xx or 4xx, generating some noise.

I also updated the ``ObsCtlRulesSetFailure` alert to not include the errors talking to the rules store API, which are also covered automatically by `ObsCtlRulesStoreServerError`. Now it only covers pre-processing that needs to be done before pushing the rules.

File: openshift/alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        sum(sum_over_time(obsctl_reloader_prom_rules_store_ops_total{job="obsctl-reloader"}[5m]))
+      ) or vector(0)
+      > 0.10
+    for: 10m
+    labels:
+      severity: critical
+  - alert: ObsCtlRulesSetFailure
+    annotations:
+      description: obsctl-reloader is failing to set rules for tenant {{ $labels.tenant }} before reaching Observatorium {{ $value | humanizePercentage }}% of the time due to {{ $labels.reason }}.
+      summary: Failing to set rules due to issue beforetalking to Observatorium.
```
Comment (philipgough) [inline openshift/alerts.yaml:21]:
```suggestion
before talking
```

**Review by thibaultmg — APPROVED** [review]

**Pull request merged**